### PR TITLE
Added LeetCode Problem No.100

### PR DESCRIPTION
**Pull request by arnabpal2022**
#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes:

### CODE REVIEW

File: leetcode/src/100.c [hunk 1]
```diff
@@ -0,0 +1,42 @@
+
```
Comment (Panquesito7) [inline leetcode/src/100.c:1]:
```suggestion
```

File: leetcode/src/100.c [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+{
+    int val;
+    struct TreeNode *left;
+    struct TreeNode *right;
+};
+
+bool isSameTree(struct TreeNode *p, struct TreeNode *q)
+{
+    if (p == NULL && q == NULL)
+        return true;
```
Comment (Panquesito7) [inline leetcode/src/100.c:19]:
```suggestion
    if (p == NULL && q == NULL) {
        return true;
    }
```

File: leetcode/src/100.c [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    else if (p && q)
+    {
+        if (p->val == q->val && isSameTree(p->left, q->left) &&
+            isSameTree(p->right, q->right))
+            return true;
+        else
+            return false;
+    }
+    else
+        return false;
```
Comment (Panquesito7) [inline leetcode/src/100.c:29]:
```suggestion
        if (p->val == q->val && isSameTree(p->left, q->left) &&
            isSameTree(p->right, q->right)) {
            return true;
        }
        else {
            return false;
        }
    }
    else {
        return false;
    }
```

**Review by hawkaii — DISMISSED** [review]
no issue seen by me .

File: leetcode/src/100.c [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+}
+
+void printTreePreOrder(struct TreeNode *p)
```
[the remark this answers is not in the document]
Comment (arnabpal2022) [reply leetcode/src/100.c:31]: yep we can skip it. good practice it would be.

**Comment by arnabpal2022** [comment]
Check it out guys. if wrong inform me.

**Review by Panquesito7 — APPROVED** [review]
Thanks! 🚀

**Review by alexpantyukhin — APPROVED** [review]

File: leetcode/src/100.c [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+struct TreeNode
+{
+    int val;
+    struct TreeNode *left;
+    struct TreeNode *right;
+};
+
+bool isSameTree(struct TreeNode *p, struct TreeNode *q)
+{
+    return (p->val == q->val && isSameTree(p->left, q->left) &&
```
Comment (alexpantyukhin) [inline leetcode/src/100.c:17]: we still have to check that `p` and `q` are not NULL before

Comment (alexpantyukhin) [reply leetcode/src/100.c:17]: @arnabpal2022 I would probably use something like this:
```c
    // both NULL - nodes are equal this case.
    if (p == NULL && q == NULL) {
        return true;
    }

    // Just one of nodes is NULL. Nodes are not equal
    if (p == NULL || q == NULL){
        return false;
    }

    // Main condition that vals are equal, and recursive check.
    return p->val == q->val
           && isSameTree(p->left, q->left)
           && isSameTree(p->right, q->right);
```

**Review by tjgurwara99 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**